crates/lair_keystore_api/src/config.rs

```diff
@@ -152,84 +154,100 @@ impl LairServerConfigInner {
+                        sodoken::SizedLockedArray::<32>::new()?;
+
+                    sodoken::argon2::blocking_argon2id(
+                        &mut *pre_secret.lock(),
+                        &*pw_hash.lock(),
+                        &salt,
+                        ops_limit,
+                        mem_limit,
+                    )?;
+
+                    Ok(pre_secret)
+                })
+                .await
+                .map_err(OneErr::new)??;
 
             // derive our context secret
             // this will be used to encrypt the context_key
-            let ctx_secret = <sodoken::BufWriteSized<32>>::new_mem_locked()?;
+            let ctx_secret =
+                Arc::new(Mutex::new(sodoken::SizedLockedArray::<32>::new()?));
             sodoken::kdf::derive_from_key(
-                ctx_secret.clone(),
+                &mut *ctx_secret.lock().lock(),
                 42,
-                *b"CtxSecKy",
-                pre_secret.clone(),
+                b"CtxSecKy",
+                &pre_secret.lock(),
             )?;
 
             // derive our signature secret
             // this will be used to encrypt the signature seed
-            let id_secret = <sodoken::BufWriteSized<32>>::new_mem_locked()?;
+            let id_secret =
+                Arc::new(Mutex::new(sodoken::SizedLockedArray::<32>::new()?));
             sodoken::kdf::derive_from_key(
-                id_secret.clone(),
+                &mut *id_secret.lock().lock(),
                 142,
-                *b"IdnSecKy",
-                pre_secret,
+                b"IdnSecKy",
+                &pre_secret.lock(),
             )?;
 
             // the context key is used to encrypt our store_file
-            let context_key = <sodoken::BufWriteSized<32>>::new_mem_locked()?;
-            sodoken::random::bytes_buf(context_key.clone()).await?;
+            let mut context_key = sodoken::SizedLockedArray::<32>::new()?;
+            sodoken::random::randombytes_buf(&mut *context_key.lock())?;
 
             // the sign seed derives our signature keypair
             // which allows us to authenticate server identity
-            let id_seed = <sodoken::BufWriteSized<32>>::new_mem_locked()?;
-            sodoken::random::bytes_buf(id_seed.clone()).await?;
+            let mut id_seed = sodoken::SizedLockedArray::<32>::new()?;
+            sodoken::random::randombytes_buf(&mut *id_seed.lock())?;
 
             // server identity encryption keypair
-            let id_pk = <sodoken::BufWriteSized<32>>::new_no_lock();
-            let id_sk = <sodoken::BufWriteSized<32>>::new_mem_locked()?;
-            use sodoken::crypto_box::curve25519xchacha20poly1305::*;
-            seed_keypair(id_pk.clone(), id_sk, id_seed.clone()).await?;
+            let mut id_pk = [0; sodoken::crypto_box::XSALSA_PUBLICKEYBYTES];
+            let mut id_sk = sodoken::SizedLockedArray::<32>::new()?;
+            sodoken::crypto_box::xsalsa_seed_keypair(
```

Comment:
Change! This is not the same libsodium function that was exposed before. Changing from `crypto_box_curve25519xchacha20poly1305_seed_keypair` to whatever algorithm `crypto_box_seed_keypair` chooses to use